src/Makefile

```
@@ -1,13 +1,19 @@
 # VERSION is the version we should download and use.
-VERSION:=$(shell git describe --match=NeVeRmAtCh --always --dirty)
+REVISION:=$(shell git describe --match=NeVeRmAtCh --always --dirty)
+RANDOM_SUFFIX:=$(shell head /dev/urandom | tr -dc A-Za-z0-9 | head -c 4)
+# if the REVISION includes `-dirty`, we append a random suffix to the version
+VERSION:=$(shell echo $(REVISION) | sed 's/-dirty/-$(RANDOM_SUFFIX)/')
 
 PROTOS := protos/io/defang/v1/fabric.pb.go protos/io/defang/v1/defangv1connect/fabric.connect.go
 
 BINARY_NAME:=defang
 GOFLAGS:=-ldflags "-X main.version=$(VERSION)"
 
-$(BINARY_NAME): test
+builds/$(BINARY_NAME)-$(VERSION): protos
 	go build -o $@ $(GOFLAGS) ./cmd/cli
+	ln -s $@ builds/$(BINARY_NAME)-latest
+
+build: builds/$(BINARY_NAME)-$(VERSION)
```

`build` is not a file target, so it's phony?
```suggestion
.PHONY: build
build: builds/$(BINARY_NAME)-$(VERSION)
```